src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -56,8 +60,8 @@ public SystemTotalCpuProvider()
 			}
 		}
 
-		protected virtual StreamReader GetProcStatAsStream()
-			=> File.Exists("/proc/stat") ? new StreamReader("/proc/stat") : null;
+		private StreamReader GetProcStatAsStream()
+			=> _procStatStreamReader ?? (File.Exists("/proc/stat") ? new StreamReader("/proc/stat") : null);
```

Comment:
Are you sure it's safe to keep this file open? I would go with open/read/close approach to minimize risk of caches interfering.

Reply:
> Are you sure it's safe to keep this file open?  

My intention is to not keep it open. But isn't that what's already happening? 

https://github.com/elastic/apm-agent-dotnet/blob/63c1a7250c693177a3165ac72ee4cfbab870e98b/src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs#L39

Reply:
Maybe to give the whole picture on this: The reason for this part is to be able to inject the content of `proc/stat` from a test.

Reply:
I see - it keeps the stream for in testing mode but opens it every time in production mode. I would say it's kinda defeats the purpose of testing :) Why not to have the same flow (stream always reopened) in both modes? You can pass stream supplier (`Func<StreamReader>`) to constructor. But that is up to you.